### PR TITLE
Execution Result Buttons

### PR DESCRIPTION
**Pull request by JoeAtHPI**
Each example gets a button which display its return value when executed in the corresponding method. When clicked, the result can be inspected.
image

Follow ups: Methods which return the receiver should display 'self'

### CODE REVIEW

File: packages/Babylonian-Core.package/BPMethodExample.class/instance/execute.st [hunk 1]
```diff
@@ -1,6 +1,11 @@
 tracing
 execute
 	
-	^ self newReceiver 
-		perform: self method selector 
-		withArguments: self newArguments
\ No newline at end of file
+	|executionResult|
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPMethodExample.class/instance/execute.st:4]: The code for recording the execution result should actually be in BlockClosure>>#bpTraceWith: tracer and should make use of `BPActiveTracer value` to ensure that the trace is constructed correctly.

Comment (codeZeilen) [reply packages/Babylonian-Core.package/BPMethodExample.class/instance/execute.st:4]: Which is the code used in bpTraceExecutionResult: :)

File: packages/Babylonian-Tests.package/BPTracingTest.class/instance/testTracingOfExecutionResult.st [hunk 2]
```diff
@@ -0,0 +1,6 @@
+tests
+testTracingOfExecutionResult
+	
+	| result |
+	result := (BPTestManuallyInstrumentedClass>>#methodAWithExamplesAndReplacementAndAssertions) bpExamples first trace.
+	self assert: (result executionResult) equals: 4
```
Comment (codeZeilen) [inline packages/Babylonian-Tests.package/BPTracingTest.class/instance/testTracingOfExecutionResult.st:6]: Too many parentheses :)

**Review by codeZeilen — CHANGES_REQUESTED** [review]

**Review by codeZeilen — CHANGES_REQUESTED** [review]
comments are inline

File: packages/Babylonian-Core.package/BPMethodExample.class/instance/execute.st [hunk 3]
```diff
@@ -1,6 +1,11 @@
...
+	executionResult := [self newReceiver 
+							perform: self method selector 
+							withArguments: self newArguments]
+								on: Error
```
Comment (codeZeilen) [inline packages/Babylonian-Core.package/BPMethodExample.class/instance/execute.st:8]: This causes the test failures as it masks the errors we actually want to record. See BlockClosure>>#bpTraceWith: for how this is done. When the code moves there you actually will not need the exception handling in this line anymore.

**Comment by JoeAtHPI** [comment]
@codeZeilen

**Comment by JoeAtHPI** [comment]
what do you mean with comments are inline? i just corrected a typo in the existing one

**Pull request merged**